crates/pathfinder/src/rpc/types.rs

```diff
@@ -297,56 +293,27 @@ pub mod reply {
     #[cfg(test)]
     impl PartialEq<jsonrpsee::core::error::Error> for ErrorCode {
         fn eq(&self, other: &jsonrpsee::core::error::Error) -> bool {
-            use jsonrpsee::core::error::Error::*;
-
-            // the interesting variant Error::Call holds the whole error value as a raw string,
-            // which looks like FailedResponse.
-            //
-            // RpcError could have more user error body, which is why there's the
-            // deny_unknown_fields, as when writing this there were no such extra types being used.
-
-            #[derive(serde::Deserialize, Debug)]
-            pub struct FailedResponse<'a> {
-                // we don't really care about this when testing; version
-                #[serde(borrow, rename = "jsonrpc")]
-                _jsonrpc: &'a serde_json::value::RawValue,
-                // don't care: request id
-                #[serde(borrow, rename = "id")]
-                _id: &'a serde_json::value::RawValue,
-                #[serde(borrow)]
-                error: RpcError<'a>,
-            }
-
-            #[derive(serde::Deserialize, Debug)]
-            #[serde(deny_unknown_fields)]
-            pub struct RpcError<'a> {
-                code: i32,
-                #[serde(borrow)]
-                message: std::borrow::Cow<'a, str>,
-            }
-
-            impl PartialEq<ErrorCode> for FailedResponse<'_> {
-                fn eq(&self, rhs: &ErrorCode) -> bool {
-                    if let Ok(lhs) = ErrorCode::try_from(self.error.code) {
-                        // make sure the error matches what we think it was; it's ... a bit extra,
-                        // but shouldn't really be an issue.
-                        &*self.error.message == lhs.as_str() && &lhs == rhs
-                    } else {
-                        false
+            use jsonrpsee::core::error::Error;
+            use jsonrpsee::types::error::CallError;
+
+            if let Error::Call(CallError::Custom(custom)) = other {
+                // this is quite ackward dance to go back to error level then come back to the
+                // custom error object. it however allows not having the json structure in two
+                // places, and leaning on ErrorObject partialeq impl.
+                let repr = match self {
+                    ErrorCode::PageSizeTooBig => {
+                        Error::from(crate::storage::EventFilterError::PageSizeTooBig(1024))
```

Comment:
Could we somehow use the PAGE_SIZE_LIMIT const instead of `1024` here?

Reply:
I was tired of typing and this being good enough, avoiding another storage deep reference but now also kind of thinking that could be avoided completely. It would require adding a newtype on the place of 1024 which would just have a getter or const which would just be the constant.

Reply:
> It would require adding a newtype on the place of 1024 which would just have a getter or const which would just be the constant.

Well, this type would have to be spelled out as well :)